Plugins/Weapon/Weapon.cpp

```diff
@@ -980,4 +1193,22 @@ int Weapon::GetLevelByClass(CNWSCreatureStats *pStats, uint32_t nClassType)
     return 0;
 }
 
+ArgumentStack Weapon::SetOneHalfStrength(ArgumentStack&& args)
+{
+    auto objectId = Services::Events::ExtractArgument<ObjectID>(args);
+
+    if(objectId == Constants::OBJECT_INVALID)
+    {
+        LOG_INFO("Invalid Object Passed into SetOneHalfStrength");
+        return Services::Events::Arguments();
+    }
+
+    auto bMulti = Services::Events::ExtractArgument<int32_t>(args);
+    if(bMulti)
+        g_plugin->GetServices()->m_perObjectStorage->Set(objectId, "ONE_HALF_STRENGTH", 1);
```

Comment:
If you want, you can change the above to `g_plugin->GetServices()->m_perObjectStorage->Set(objectId, "ONE_HALF_STRENGTH", 1, true);` and it'll persist the value when the item gets saved, so you only have to set it once.